spek-extensions/robolectric/src/main/kotlin/org/spekframework/spek2/robolectric/RobolectricInstanceFactory.kt

```diff
@@ -0,0 +1,274 @@
+package org.spekframework.spek2.robolectric
+
+import org.robolectric.ConfigMerger
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.android.AndroidInterceptors
+import org.robolectric.annotation.Config
+import org.robolectric.internal.AndroidConfigurer
+import org.robolectric.internal.BuckManifestFactory
+import org.robolectric.internal.DefaultManifestFactory
+import org.robolectric.internal.GradleManifestFactory
+import org.robolectric.internal.ManifestFactory
+import org.robolectric.internal.ManifestIdentifier
+import org.robolectric.internal.MavenManifestFactory
+import org.robolectric.internal.SandboxFactory
+import org.robolectric.internal.SdkConfig
+import org.robolectric.internal.SdkEnvironment
+import org.robolectric.internal.bytecode.ClassHandler
+import org.robolectric.internal.bytecode.InstrumentationConfiguration
+import org.robolectric.internal.bytecode.Interceptor
+import org.robolectric.internal.bytecode.Interceptors
+import org.robolectric.internal.bytecode.SandboxConfig
+import org.robolectric.internal.bytecode.ShadowMap
+import org.robolectric.internal.bytecode.ShadowWrangler
+import org.robolectric.internal.dependency.CachedDependencyResolver
+import org.robolectric.internal.dependency.DependencyResolver
+import org.robolectric.internal.dependency.LocalDependencyResolver
+import org.robolectric.internal.dependency.PropertiesDependencyResolver
+import org.robolectric.manifest.AndroidManifest
+import org.robolectric.res.Fs
+import org.robolectric.util.Logger
+import org.robolectric.util.ReflectionHelpers
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.lifecycle.InstanceFactory
+import java.io.File
+import java.io.IOException
+import java.security.SecureRandom
+import java.util.ArrayList
+import java.util.HashMap
+import java.util.Properties
+import kotlin.reflect.KClass
+
+object RobolectricInstanceFactory : InstanceFactory {
+    private val appManifestsCache = HashMap<ManifestIdentifier, AndroidManifest>()
+
+    @Transient private var dependencyResolver: DependencyResolver? = null
```

Comment:
why do we need `@Transient`?

Reply:
dependecies are cached by cacheddependecy resolve and it is composed there.  Robolectric also declare it as transient.